packages/core/src/lib/button.svelte

```diff
@@ -41,50 +41,55 @@ export let width: 'full' | 'default' = 'default';
 
 <button
   {type}
-  aria-label={variant === 'icon' ? label : undefined}
+  aria-label={variant === 'icon' || variant === 'icon-danger'
+    ? label
+    : undefined}
   aria-disabled={disabled}
   {disabled}
   {title}
-  class={cx('mx-auto whitespace-nowrap', {
+  class={cx('whitespace-nowrap', {
     'flex w-full': width === 'full',
-    'inline-flex': width !== 'full' && variant !== 'icon',
+    'inline-flex':
+      width !== 'full' && variant !== 'icon' && variant !== 'icon-danger',
     'h-[30px] w-[30px] text-gray-6 hover:text-gray-7 active:text-gray-8':
       variant === 'icon',
-    'px-3': !icon && variant !== 'icon',
-    'pl-2 pr-3': icon && variant !== 'icon',
-    'items-center justify-center gap-1.5 border px-3 py-1.5 text-xs':
-      variant !== 'icon',
+    'h-[30px] w-[30px] text-gray-6 hover:text-danger-dark active:text-danger-dark':
```

Comment:
Leaving ugly for now but going to refactor this logic in future.

Reply:
maybe `<IconButton>` and `<Button>` need to be separated, conditionals are getting kinda cray cray

Reply:
will do as a follow up